File: service/testsupport/src/main/java/com/github/goldy1992/mp3player/service/TestMediaPlaybackServiceInjector.java
package com.github.goldy1992.mp3player.service;

import com.github.goldy1992.mp3player.commons.ComponentClassMapper;
import com.github.goldy1992.mp3player.service.MediaPlaybackService;
import com.github.goldy1992.mp3player.service.dagger.components.ServiceComponent;
import com.github.goldy1992.mp3player.service.dagger.components.TestServiceComponent;
import com.github.goldy1992.mp3player.service.dagger.components.DaggerTestServiceComponent;

public class TestMediaPlaybackServiceInjector extends MediaPlaybackService {

    @Override
    public void onCreate() {
        initialiseDependencies();
        super.onCreate();
    }

    /**
     * TO BE CALLED BEFORE SUPER CLASS
     */
    @Override
    protected void initialiseDependencies() {
       ServiceComponent component =  DaggerTestServiceComponent
                .factory()
                .create(getApplicationContext(), this, "MEDIA_PLYBK_SRVC_WKR", new ComponentClassMapper.Builder().build());
                component.inject(this);
    }
}

File: service/testsupport/src/main/java/com/github/goldy1992/mp3player/service/TestMediaPlaybackServiceInjector.kt
package com.github.goldy1992.mp3player.service

import com.github.goldy1992.mp3player.commons.ComponentClassMapper
import com.github.goldy1992.mp3player.service.dagger.components.DaggerTestServiceComponent

class TestMediaPlaybackServiceInjector : MediaPlaybackService() {
    override fun onCreate() {
        initialiseDependencies()
        super.onCreate()
    }

    /**
     * TO BE CALLED BEFORE SUPER CLASS
     */
    override fun initialiseDependencies() {
        val component = DaggerTestServiceComponent
                .factory()
                .create(applicationContext, this, "MEDIA_PLYBK_SRVC_WKR", ComponentClassMapper.Builder().build())
        component!!.inject(this)
    }
}